src/lib/dnssd/platform/Dnssd.h

```diff
@@ -205,6 +210,32 @@ CHIP_ERROR ChipDnssdFinalizeServiceUpdate();
 CHIP_ERROR ChipDnssdBrowse(const char * type, DnssdServiceProtocol protocol, chip::Inet::IPAddressType addressType,
                            chip::Inet::InterfaceId interface, DnssdBrowseCallback callback, void * context);
 
+/**
+ * This function stops service browsing.
+ *
+ * The type, protocol, addressType and interface parameters must match the ones used
+ * in the call to the `ChipDnssdBrowse` function. Alternatively, one shall be able
+ * to use "wildcard" values for the type (nullptr), protocol (kDnssdProtocolUnknown),
+ * addressType (IPAddressType::kAny) and interface (InterfaceId::Null()) to discard
+ * matches for particular browsing parameters.
+ *
+ * Please note, that when stopping mDNS browsing, it is REQUIRED to call the callback
+ * passed to the `ChipDnssdBrowse` function with an error code or the CHIP_END_OF_INPUT
+ * code. This will allow the mDNS module to release its resources.
+ *
+ * @param[in] type        The service type.
+ * @param[in] protocol    The service protocol.
+ * @param[in] addressType The protocol version of the IP address.
+ * @param[in] interface   The interface on which queries are sent.
+ *
+ * @retval CHIP_NO_ERROR                The stop browse succeeds or no browse for
+ *                                      given parameters is in progress.
+ * @retval Error code                   The stop browse fails.
+ *
+ */
+CHIP_ERROR ChipDnssdStopBrowse(const char * type, DnssdServiceProtocol protocol, chip::Inet::IPAddressType addressType,
```

Comment:
Is there a public API that actually ends up calling this?  Or is that planned to be a followup?

Reply:
Yes, I'm planing to do a follow up for that, but to be honest I've faced some issues. I'm not sure right now whether start/stop approach is the easiest way of achieving the goal that I've set for myself - I'd like that Tizen will be able to discover more than one Matter device. I'm still doing some research what approach would be better or how to apply my current proposal, so please do not close this conversation for now.

Reply:
I've dropped this new ChipDnssdStopBrowse API in favor of #21553